File: 0735-asteroid-collision/0735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> s = new Stack<>();

        s.push(asteroids[0]);

        int n = asteroids.length;

        for(int i = 1; i < n ; i++){
            boolean flag = true;

            while(!s.isEmpty() && asteroids[i] < 0 && s.peek() > 0){
                if(Math.abs(asteroids[i]) > s.peek()){
                    s.pop();
                    continue;
                } else if(Math.abs(asteroids[i]) == s.peek()){
                    s.pop();
                    flag = false;
                    break;
                } else{
                    flag = false;
                    break;
                }
            }

            if(flag){
                s.push(asteroids[i]);
            }
        }

        int[] ans = new int[s.size()];

        for(int i = ans.length - 1 ; i >= 0 ; i--){
            ans[i] = s.pop();
        }

        return ans;
    }
}

File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        stack = []

        for i in range(0 , len(asteroids)):
            flag = True

            while len(stack) != 0 and asteroids[i] < 0 and stack[-1] > 0 :
                if abs(asteroids[i]) > stack[-1] : 
                    stack.pop()
                    continue

                elif abs(asteroids[i]) == stack[-1] :
                    stack.pop()
                    flag = False
                    break

                else :
                    flag = False
                    break

            if flag :
                stack.append(asteroids[i])

        return stack